src/measure_theory/tactic.lean

```diff
@@ -0,0 +1,181 @@
+/-
+Copyright (c) 2021 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne
+-/
+import tactic.auto_cases
+import tactic.tidy
+import tactic.with_local_reducibility
+import tactic.show_term
+import measure_theory.measure_space_def
+/-!
+# Tactics for measure theory
+
+Currently we have one domain-specific tactic for measure theory: `measurability`.
+
+This tactic is to a large extent a copy of the `continuity` tactic by Reid Barton.
+-/
+
+/-!
+### `measurability` tactic
+
+Automatically solve goals of the form `measurable f`, `ae_measurable f μ` and `measurable_set s`.
+
+Mark lemmas with `@[measurability]` to add them to the set of lemmas
+used by `measurability`. Note: `to_additive` doesn't know yet how to
+copy the attribute to the additive version.
+-/
+
+/-- User attribute used to mark tactics used by `measurability`. -/
+@[user_attribute]
+meta def measurability : user_attribute :=
+{ name := `measurability,
+  descr := "lemmas usable to prove (ae)-measurability" }
+
+/- Mark some measurability lemmas already defined in `measure_theory.measurable_space_0` and
+`measure_theory.measure_space_0` -/
+attribute [measurability]
+  measurable_id
+  measurable_const
+  ae_measurable_const
+  ae_measurable.measurable_mk
+  measurable_set.empty
+  measurable_set.univ
+  measurable_set.compl
+  subsingleton.measurable_set
+  measurable_set.Union
+  measurable_set.Inter
+  measurable_set.Union_Prop
+  measurable_set.Inter_Prop
+  measurable_set.union
+  measurable_set.inter
+  measurable_set.diff
+  measurable_set.ite
+  measurable_set.disjointed
+  measurable_set.const
+  measurable_set.insert
+  measurable_set_eq
+  set.finite.measurable_set
+  finset.measurable_set
+  set.countable.measurable_set
+  measurable_space.measurable_set_top
+
+-- As we will be using `apply_rules` with `md := semireducible`,
+-- we need another version of `measurable_id`.
+@[measurability] lemma measurable_id' {α : Type*} [measurable_space α] : measurable (λ a : α, a) :=
```

Comment:
I think you can move this one close to `measurable_id`, as this form can prove useful generally.